src/OpenLoco/Windows/KeyboardShortcuts.cpp

```diff
@@ -97,13 +99,66 @@ namespace OpenLoco::Ui::Windows::KeyboardShortcuts
         self->draw(context);
     }
 
+    static void getBindingString(uint32_t keyCode, char* buffer, const size_t bufferLength)
+    {
+        static const std::unordered_map<uint32_t, string_id> keysToString = { {
+            { SDLK_BACKSPACE, StringIds::keyboard_backspace },
+            { SDLK_TAB, StringIds::keyboard_tab },
+            { SDLK_RETURN, StringIds::keyboard_return },
+            { SDLK_PAUSE, StringIds::keyboard_pause },
+            { SDLK_CAPSLOCK, StringIds::keyboard_caps },
+            { SDLK_ESCAPE, StringIds::keyboard_escape },
+            { SDLK_SPACE, StringIds::keyboard_spacebar },
+            { SDLK_PAGEUP, StringIds::keyboard_pageup },
+            { SDLK_PAGEDOWN, StringIds::keyboard_pagedown },
+            { SDLK_END, StringIds::keyboard_end },
+            { SDLK_HOME, StringIds::keyboard_home },
+            { SDLK_LEFT, StringIds::keyboard_left },
+            { SDLK_UP, StringIds::keyboard_up },
+            { SDLK_RIGHT, StringIds::keyboard_right },
+            { SDLK_DOWN, StringIds::keyboard_down },
+            { SDLK_INSERT, StringIds::keyboard_insert },
+            { SDLK_DELETE, StringIds::keyboard_delete },
+            { SDLK_KP_1, StringIds::keyboard_numpad_1 },
+            { SDLK_KP_2, StringIds::keyboard_numpad_2 },
+            { SDLK_KP_3, StringIds::keyboard_numpad_3 },
+            { SDLK_KP_4, StringIds::keyboard_numpad_4 },
+            { SDLK_KP_5, StringIds::keyboard_numpad_5 },
+            { SDLK_KP_6, StringIds::keyboard_numpad_6 },
+            { SDLK_KP_7, StringIds::keyboard_numpad_7 },
+            { SDLK_KP_8, StringIds::keyboard_numpad_8 },
+            { SDLK_KP_9, StringIds::keyboard_numpad_9 },
+            { SDLK_KP_0, StringIds::keyboard_numpad_0 },
+            { SDLK_KP_DIVIDE, StringIds::keyboard_numpad_divide },
+            { SDLK_KP_ENTER, StringIds::keyboard_numpad_enter },
+            { SDLK_KP_MINUS, StringIds::keyboard_numpad_minus },
+            { SDLK_KP_MULTIPLY, StringIds::keyboard_numpad_multiply },
+            { SDLK_KP_PERIOD, StringIds::keyboard_numpad_period },
+            { SDLK_KP_PLUS, StringIds::keyboard_numpad_plus },
+            { SDLK_NUMLOCKCLEAR, StringIds::keyboard_numlock },
+            { SDLK_SCROLLLOCK, StringIds::keyboard_scroll },
+        } };
+
+        auto match = keysToString.find(keyCode);
+        if (match != keysToString.end())
+        {
+            StringManager::formatString(buffer, match->second);
+        }
+        else
+        {
+            const char* sdlBuffer = SDL_GetKeyName(keyCode);
+            strncpy(buffer, sdlBuffer, bufferLength);
```

Comment:
Fedora CI complains about this line. I'm not sure how to resolve this yet:
```
../src/OpenLoco/Windows/KeyboardShortcuts.cpp:150:20: error: 'char* strncpy(char*, const char*, size_t)'
specified bound 128 equals destination size [-Werror=stringop-truncation]
  150 |             strncpy(buffer, sdlBuffer, bufferLength);
      |             ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
strncpy needs to copy n-1 chars to account for the null terminator:

https://en.cppreference.com/w/cpp/string/byte/strncpy
#

Reply:
You dont need to use a c string anyway. Pass back a std::string from the function and use .c_str() in the format arguments.

Reply:
> strncpy needs to copy n-1 chars to account for the null terminator:

You're right. In my mind, the third argument was just to denote the length of the dest buffer. Now I wonder why we've never seen this error before.

> You dont need to use a c string anyway. Pass back a std::string from the function and use .c_str() in the format arguments.

The reason I went with a C string is because our formatString function still only accepts a char buffer.